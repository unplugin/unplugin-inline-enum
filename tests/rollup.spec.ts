import path from 'node:path'
import { rollupBuild } from '@sxzz/test-utils'
import Oxc from 'unplugin-oxc/rollup'
import { expect, test } from 'vitest'
import UnpluginInlineEnum from '../src/rollup'

test('rollup', async () => {
  const { snapshot } = await rollupBuild(
    path.resolve(__dirname, 'fixtures/main.ts'),
    [
      UnpluginInlineEnum({
        scanDir: path.resolve(__dirname, 'fixtures'),
        scanMode: 'fs',
      }),
      Oxc(),
      {
        name: 'test:mod-options',
        options(options) {
          options.treeshake = 'smallest'
        },
      },
    ],
  )
  expect(snapshot).toMatchSnapshot()
})
