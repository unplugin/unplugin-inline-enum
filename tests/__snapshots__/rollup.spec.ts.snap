// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rollup 1`] = `
"// main.js
const values = [
	"foo",
	100,
	4
];

console.log("foo", 100, 4);
console.log(values);
"
`;
